web-common/src/features/dashboards/time-dimension-details/getTDDExportArgs.ts

```diff
@@ -0,0 +1,104 @@
+import { MetricsExplorerEntity } from "@rilldata/web-common/features/dashboards/stores/metrics-explorer-entity";
+import {
+  TimeControlState,
+  useTimeControlStore,
+} from "@rilldata/web-common/features/dashboards/time-controls/time-control-store";
+import {
+  mapComparisonTimeRange,
+  mapTimeRange,
+} from "@rilldata/web-common/features/dashboards/time-controls/time-range-mappers";
+import {
+  V1MetricsViewAggregationMeasure,
+  V1MetricsViewAggregationRequest,
+  V1MetricsViewSpec,
+} from "@rilldata/web-common/runtime-client";
+import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+import { derived, get, Readable } from "svelte/store";
+
+export function getTDDExportArgs(
+  ctx: StateManagers,
+): Readable<V1MetricsViewAggregationRequest | undefined> {
+  return derived(
+    [
+      ctx.metricsViewName,
+      ctx.dashboardStore,
+      useTimeControlStore(ctx),
+      useMetricsView(ctx),
+    ],
+    ([metricsViewName, dashboardState, timeControlState, metricsView]) =>
+      getTDDAggregationRequest(
+        metricsViewName,
+        dashboardState,
+        timeControlState,
+        metricsView.data,
+      ),
+  );
+}
+
+export function getTDDAggregationRequest(
+  metricsViewName: string,
+  dashboardState: MetricsExplorerEntity,
+  timeControlState: TimeControlState,
+  metricsView: V1MetricsViewSpec | undefined,
+): undefined | V1MetricsViewAggregationRequest {
+  if (
+    !metricsView ||
+    !timeControlState.ready ||
+    !dashboardState.tdd.expandedMeasureName
+  )
+    return undefined;
+
+  const timeRange = mapTimeRange(timeControlState, metricsView);
+  if (!timeRange) return undefined;
+
+  const comparisonTimeRange = mapComparisonTimeRange(
+    dashboardState,
+    timeControlState,
+    timeRange,
+  );
+
+  const measures: V1MetricsViewAggregationMeasure[] = [
+    { name: dashboardState.tdd.expandedMeasureName },
+  ];
+  const { sortMeasureName, measures: apiMeasures } = getMeasureNames(
+    dashboardState.tdd.expandedMeasureName,
+    dashboardState.dashboardSortType,
+    !!comparisonTimeRange?.start &&
+      !!comparisonTimeRange?.end &&
+      !!timeControlState.selectedComparisonTimeRange,
+  );
+  measures.push(...apiMeasures);
+
+  // CAST SAFETY: exports are only available in TDD when a comparison dimension is selected
+  const dimensionName = dashboardState.selectedComparisonDimension as string;
+
+  return {
+    instanceId: get(runtime).instanceId,
+    metricsView: metricsViewName,
+    dimensions: [
+      { name: dimensionName },
+      {
+        name: metricsView.timeDimension ?? "",
+        timeGrain: dashboardState.selectedTimeRange?.interval,
+        timeZone: dashboardState.selectedTimezone,
+      },
+    ],
+    measures,
+    timeRange,
+    ...(comparisonTimeRange ? { comparisonTimeRange } : {}),
+    sort: [
+      {
+        name: sortMeasureName,
+        desc: dashboardState.sortDirection === SortDirection.DESCENDING,
+      },
+    ],
```

Comment:
For TDD, I'd expect that the data is displayed and sorted by [time, dimension], e.g.:
- 1/1/24, Canada, 1000
- 1/1/24, USA, 1500
- 1/2/24, Canada, 2000
- 1/2/24, USA, 1500

Here is the current display and sort behavior that confused me:
![image](https://github.com/rilldata/rill/assets/14206386/ec322e9c-178c-4372-9357-6e526a054cce)

Reply:
That export looks like it's diverging from how Export in the UI looks like. Export in the UI would also pivot on time